contracts/Honey.sol

```diff
@@ -177,4 +183,17 @@ contract Honey is IERC20 {
 
         _transfer(from, to, value);
     }
+
+    function registerTokenOnL2(
+        address l2CustomTokenAddress,
+        uint256 maxSubmissionCost,
+        uint256 maxGas,
+        uint256 gasPriceBid
+    ) external {
+        // TODO: Do we need this?
+    }
+
+    function registerWithGatewayRouter(uint256 _maxGas, uint256 _gasPriceBid, uint256 _maxSubmissionCost) external {
+        gatewayRouter.setGateway(gateway, _maxGas, _gasPriceBid, _maxSubmissionCost);
```

Comment:
Note that we need it for the same reason as above.

Reply:
This is definitely necessary, unless the arb admins will add our CustomGateway manually. Or if it turns out you actually you don't need to strictly register with the GatewayRouter to use the Arbitrum deployed CustomGateway but that would undermine the point of the GatewayRouter so I doubt that's the case.